drivers/soundwire/bus.c

```diff
@@ -500,10 +500,21 @@ int sdw_nread(struct sdw_slave *slave, u32 addr, size_t count, u8 *val)
 {
 	int ret;
 
-	ret = pm_runtime_get_sync(&slave->dev);
-	if (ret < 0 && ret != -EACCES) {
-		pm_runtime_put_noidle(&slave->dev);
-		return ret;
+	/*
+	 * sdw_read(), or its regmap equivalent, may be called from a
+	 * resume function, which prevents us from using
+	 * pm_runtime_get_sync() here.  Instead we first increase the
+	 * reference count, then check the status.  If the device is
+	 * resuming already or is active, we can directly read the data
+	 */
+	pm_runtime_get_noresume(&slave->dev);
+
+	if (pm_runtime_status_suspended(&slave->dev)) {
+		ret = pm_runtime_resume(&slave->dev);
```

Comment:
@plbossart just out of cuioristy why not just pm_runtime_get_sync() here?

Reply:
@ranj063 because we can't call pm_runtime_get_sync() from the resume routine. that leads to a deadlock.